gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -638,242 +654,321 @@ _gst_tensor_filter_transform_validate (GstBaseTransform * trans,
-  gboolean allocate_in_invoke, in_flexible, out_flexible;
-  gboolean need_profiling;
-  gsize expected, hsize;
-
-  GstTensorMetaInfo in_meta[NNS_TENSOR_SIZE_LIMIT];
-  GstTensorMetaInfo out_meta[NNS_TENSOR_SIZE_LIMIT];
-
-  GstMemory *mem;
-
-  /* 0. Check all properties. */
-  GstFlowReturn retval = _gst_tensor_filter_transform_validate (trans, inbuf,
-      outbuf);
-  if (retval != GST_FLOW_OK)
-    return retval;
+  guint i;
+  gsize hsize;
+  FilterTransformData *trans_data = NULL;
 
-  allocate_in_invoke = gst_tensor_filter_allocate_in_invoke (priv);
+  trans_data = g_new0 (FilterTransformData, 1);
 
-  in_flexible =
-      gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SINK_PAD (trans));
-  out_flexible =
-      gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SRC_PAD (trans));
-
-  if (priv->prop.invoke_dynamic && !out_flexible) {
-    ml_loge
-        ("Dynamic Invoke of tensor filter is activated but the output of tensor filter is static tensors. Currently, only flexible tensors is supported as output of dynamic invoke. If you don't want to dynamic invoke, remove the invoke-dynamic option of tensor filter.");
-    return GST_FLOW_ERROR;
+  if (!trans_data) {
+    ml_loge ("Failed to allocate memory for internal data of tensor filter transform input data.");
+    return NULL;
   }
 
-  /* 1. Get all input tensors from inbuf. */
-  /* Internal Logic Error or GST Bug (sinkcap changed!) */
-  num_tensors = gst_tensor_buffer_get_count (inbuf);
+  trans_data->num_tensors = gst_tensor_buffer_get_count (buf);
 
-  for (i = 0; i < num_tensors; i++) {
-    in_mem[i] = gst_tensor_buffer_get_nth_memory (inbuf, i);
-    if (!gst_memory_map (in_mem[i], &in_info[i], GST_MAP_READ)) {
+  for (i = 0; i < trans_data->num_tensors; i++) {
+    trans_data->mem[i] = gst_tensor_buffer_get_nth_memory (buf, i);
+    if (!gst_memory_map (trans_data->mem[i], &trans_data->info[i],
+            GST_MAP_READ)) {
       ml_logf_stacktrace
           ("gst_tensor_filter_transform: For the given input buffer, tensor-filter (%s : %s) cannot map input memory from the buffer for reading. The %u-th memory chunk (%u-th tensor) has failed for memory map.\n",
           prop->fwname, TF_MODELNAME (prop), i, i);
-      goto mem_map_error;
+      g_free (trans_data);
+      return NULL;
     }
 
+    trans_data->is_flexible =
+        gst_tensor_pad_caps_is_flexible (GST_BASE_TRANSFORM_SINK_PAD (trans));
     hsize = 0;
-    if (in_flexible) {
-      gst_tensor_meta_info_parse_header (&in_meta[i], in_info[i].data);
-      hsize = gst_tensor_meta_info_get_header_size (&in_meta[i]);
-      gst_tensor_meta_info_convert (&in_meta[i], &prop->input_meta.info[i]);
+    if (trans_data->is_flexible) {
+      gst_tensor_meta_info_parse_header (&trans_data->meta[i],
+          trans_data->info[i].data);
+      hsize = gst_tensor_meta_info_get_header_size (&trans_data->meta[i]);
+      gst_tensor_meta_info_convert (&trans_data->meta[i],
+          &prop->input_meta.info[i]);
```

Comment:
The current code is fine 👍 , but what if we make this part into a function like below?
```
static gsize
gst_tensor_filter_get_header_size (GstTensorFilterPrivate * priv, FilterTransformData * trans_data)
{
  gsize header_size = 0;
  GstTensorFilterProperties *prop = &priv->prop;
  if (trans_data->in_flexible) {
      gst_tensor_meta_info_parse_header (&trans_data->in_meta[I], trans_data->in_info[i].data);
      header_size = gst_tensor_meta_info_get_header_size (&trans_data->in_meta[i]);
      gst_tensor_meta_info_convert (&trans_data->in_meta[I], &prop->input_meta.info[i]);
  }
  return header_size;
}  
```

Reply:
Updated as you suggested. Thanks.